### PR TITLE
[torchlib] Fix aten::diagonal

### PR DESCRIPTION
**Pull request by justinchuby**
Turn aten::diagonal as trace only and fix its logic by explicitly converting python constants to onnx constants. This was needed because the exporter logic was not handling the type conversion correctly (yet)

### CODE REVIEW

**Review by titaiwangms — APPROVED** [review]

**Comment by fatcat-z** [comment]
If possible, it'd better paste the error we saw into description so we know what kind of logic was fixed.

**Comment by justinchuby** [comment]
> If possible, it'd better paste the error we saw into description so we know what kind of logic was fixed.

Done, thanks.

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -2550,19 +2550,10 @@ def aten_diagonal(self: TReal, offset: int = 0, dim1: int = 0, dim2: int = 1) ->
...
-
-
-@torch_op("aten::diagonal", private=True, traceable=True)
-def _aten_diagonal_onnx(
-    self: TTensor, offset: int, dim1: int, dim2: int, perm: Sequence[int], axes: Sequence[int]
-) -> TTensor:
-    neg_1 = op.Constant(value_ints=[-1])
-    dim1_size = op.Reshape(op.Gather(op.Shape(self), dim1), neg_1)  # row
-    dim2_size = op.Reshape(op.Gather(op.Shape(self), dim2), neg_1)  # col
+    dim1_size = op.Shape(self, end=dim1, start=dim1 + 1)  # row
```
Comment (gramalingam) [inline onnxscript/function_libs/torch_lib/ops/core.py:2553]: Can `dim1` be negative? Especially -1? Same for `dim2`.

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:2553]: ah it can. Thanks for pointing this out!

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py]: Reverted to use gather. Added a comment

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py]: Hmm but its handled in 
image

**Pull request merged**